### PR TITLE
Sets up initial charge analytics with Segment

### PR DESCRIPTION
**Pull request by discorick**
V1 to get the data rolling in, this creates an identity per board, with a user key for each user on that board. This should allow us to create interesting data about individual boards, along with insights about usage across different boards.

### CODE REVIEW

File: app/jobs/analytics/identify_user_job.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+module Analytics
+  class IdentifyUserJob < AnalyticsJob
+
+    def payload(params)
+      {
+        user_id: params['owner']['id'],
+        traits: {
+          email: params['email'] || "",
+          user: params['user'],
+          account: params['account_id'] || ""
```
Comment (rauhryan) [inline app/jobs/analytics/identify_user_job.rb]: please add the account type, I would like to know whether the owner is a user or an org to correlate whether we should get rid of the user plan

File: config/initializers/analytics.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+if ENV["ANALYTICS_KEY"]
```
Comment (rauhryan) [inline config/initializers/analytics.rb]: Why different env variable for Segment?

File: app/jobs/analytics/identify_user_job.rb (hunk 1 above)
Comment (discorick) [inline app/jobs/analytics/identify_user_job.rb]: Ok - still working on what these payloads should actually be, waiting on Segment to respond.

File: config/initializers/analytics.rb (hunk 2 above)
Comment (discorick) [inline config/initializers/analytics.rb]: I didn't realize we already had a `SEGMENTIO_KEY` var.

File: app/jobs/analytics/identify_user_job.rb (hunk 1 above)
Comment (rauhryan) [inline app/jobs/analytics/identify_user_job.rb]: specifically interested in params['user']['type']

**Comment by rauhryan** [comment]
Let's talk more about extending the job resolver and queuer

I want all the analytics jobs to be in the SaaS namespace so we can exclude it from enterprise

**Comment by discorick** [comment]
Ok ill hold off on further development.

**Comment by rauhryan** [comment]
You're fine to keep working.

I just want to move things around before we ship

So keep doing what you're doing for now

**Comment by rauhryan** [comment]
Yo @discorick check out the new hotness

I'm getting the hang of this analytics game son!

**Comment by rauhryan** [comment]
@discorick :shipit:?

It's a good start to start populating `user` -> `repo owner`

**Comment by discorick** [comment]
:shipit:

**Pull request merged**